### PR TITLE
TEST: Add Cornerstone Tools@next

### PR DESCRIPTION
**Pull request by swederik**
*Do not merge!!!!*

TODO:
- Switch the OHIF Tool Manager to use the Cornerstone Tools 3.x API

- https://github.com/OHIF/Viewers/blob/c8f242d5f54aa7841b7943f88d5acf522daa211e/Packages/ohif-viewerbase/client/lib/toolManager.js

- https://github.com/cornerstonejs/cornerstoneTools/projects/1

### CODE REVIEW

**Comment by swederik** [comment]
must declare valid version of dependency:
8:10:51 PM: cornerstone-tools@~3.0.0-b.0

File: StandaloneViewer/StandaloneViewer/client/components/toolbarSection/toolbarSection.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ Template.toolbarSection.helpers({
         });
 
         extraTools.push({
-            id: 'dragProbe',
+            id: 'probe',
```
Comment (swederik) [inline StandaloneViewer/StandaloneViewer/client/components/toolbarSection/toolbarSection.js]: We intend to keep dragProbe, so switch this back when that tool has been migrated.

File: StandaloneViewer/StandaloneViewer/client/components/toolbarSection/toolbarSection.js [hunk 2]
```diff
@@ -126,7 +126,7 @@ Template.toolbarSection.helpers({
         const buttonData = [];
 
         buttonData.push({
-            id: 'stackScroll',
+            id: 'stackScrollMouseWheel',
```
Comment (swederik) [inline StandaloneViewer/StandaloneViewer/client/components/toolbarSection/toolbarSection.js]: This button enables left-click dragging, not mousewheel, so this change makes no sense.

File: Packages/ohif-viewerbase/client/lib/toolManager.js [hunk 3]
```diff
@@ -12,51 +12,13 @@ import { textMarkerUtils } from './textMarkerUtils';
 import { isTouchDevice } from './helpers/isTouchDevice';
 
 let defaultTool = {
-    left: 'wwwc',
+    left: 'wwwcTool',
```
Comment (swederik) [inline Packages/ohif-viewerbase/client/lib/toolManager.js]: Why the change?

File: Packages/ohif-viewerbase/client/lib/hotkeyUtils.js [hunk 4]
```diff
@@ -110,12 +110,13 @@ Meteor.startup(function() {
...
-        dragProbe: 'Pixel Probe',
+        probe: 'Pixel Probe',
+        stackScroll: 'Stack Scroll',
         ellipticalRoi: 'Elliptical ROI',
         rectangleRoi: 'Rectangle ROI',
         magnify: 'Magnify',
-        annotate: 'Annotate',
-        stackScroll: 'Scroll Stack',
+        arrowAnnotate: 'Annotate',
+        stackScrollMouseWheel: 'Scroll Stack Mouse Wheel',
```
Comment (swederik) [inline Packages/ohif-viewerbase/client/lib/hotkeyUtils.js]: Not necessary

**Comment by evren217** [comment]
Here is more todos (besides the ones in the code):
1. Mouse wheel does not scroll images
2. Clicking or dragging on scrollbar does not go to image
3. Tool configurations (e.g. shadowConfig) are not added
4. Right-click on an annotation does not display context-menu in order to delete the annotation
5. Tools do not work anymore after displaying download dialog

**Comment by JamesAPetts** [comment]
An observation: tools aren't set passive by default by `cornerstoneTools`, only after they have been activated then switched. So if you load in some `toolData` from an external source, you have to set the tool `passive` (or `enabled`) manually if it hasn't been used yet in order to get the data to render.

**Comment by dannyrb** [comment]
Is this still in-progress?

**Comment by evren217** [comment]
There are still a few todos:
- [ ] Lesion tracker tools (bi-directional, non-target, etc.) do not work (those tools need to be migrated to cornerstone tools v3)
- [ ] Measurements are not displayed on the initial draw of elliptical and rectangular rois

**Comment by dannyrb** [comment]
> Measurements are not displayed on the initial draw of elliptical and rectangular rois

I believe this was fixed in a recent `cornerstone-tools` release. If you're still having issues, let me know?

> Lesion tracker tools (bi-directional, non-target, etc.) do not work (those tools need to be migrated to cornerstone tools v3)

I'm less helpful here, but feel free to ping me if you think I might be of assistance.

Part of me had wondered if this PR/effort had been abandoned as the `react` branch was built from the on-set using v3.

**Comment by evren217** [comment]
I rebased this PR on top of master and bumped cornerstone-tools to 3.6.0

Here is the updated todos:
- [ ] Lesion tracker tools (bi-directional, non-target, etc.) do not work (those tools need to be migrated to cornerstone tools v3)
- [ ] Measurement values (HU) are not displayed for elliptical and rectangular rois on measurement table

**Comment by galelis** [comment]
@evren217 

- [ ] Bidirectional not working is due to a case sensitive issue. Its configured inside ohif-core as 'bidirectional' and in the app as 'Bidirectional'
- [ ] non-target is not migrated, indeed. 
- [ ] ellipticalRoi and rectangleRoi is not working because there is new changes into the cornerstoneTools to save the data in a different place (if I am not wrong) 
Its now stored inside data.cachedStats
https://github.com/cornerstonejs/cornerstoneTools/blob/master/src/tools/annotation/EllipticalRoiTool.js#L282

**Comment by dannyrb** [comment]
> ellipticalRoi and rectangleRoi is not working because there is new changes into the cornerstoneTools to save the data in a different place (if I am not wrong) Its now stored inside data.cachedStats.

If that was a breaking change in `tools`, feel free to create an issue with the expected behavior and I'll create a patch.

**Comment by evren217** [comment]
@dannyrb I think that should be fixed as a part of this PR with the schema changes for elliptical roi [here](https://github.com/OHIF/Viewers/blob/cornerstone-tools-next/Packages/ohif-measurement-table/client/schema/ellipticalRoi.js#L69-L79) and rectangular roi [here](https://github.com/OHIF/Viewers/blob/cornerstone-tools-next/Packages/ohif-measurement-table/client/schema/rectangleRoi.js#L69-L79)

**Comment by evren217** [comment]
@dannyrb regarding to the bidirectional tool, there was an issue with setting tool name in cornerstone tools. Could you please take a look at [this old PR](https://github.com/cornerstonejs/cornerstoneTools/pull/747)? Let me or @galelis know if you have any question.

**Comment by swederik** [comment]
Closing this because we are not supporting 1.x. If someone wants to take this further, leave a comment.
